src/generateConfig.js

```diff
@@ -130,16 +133,27 @@ function aggregateDeps ({ packageModules, moduleIdToPackageName }) {
   return depsArray
 }
 
-function createSpy (onData, onEnd) {
+function createSpy (onData, writeAutoConfigDebug, onEnd) {
+  const allDeps = {}
   return through.obj((data, _, cb) => {
     // give data to observer fn
     onData(data)
+    // format data obj with ID as key
+    if (writeAutoConfigDebug) {
+      const metaData = clone(data)
+      allDeps[metaData.id] = metaData
+    }
     // pass the data through normally
     cb(null, data)
   }, (cb) => {
     // call flush observer
     onEnd()
-    // end normally
+    // if writeAutoConfigDebug option enabled, dump allDeps to a file
+    if (writeAutoConfigDebug) {
+      const serialized = jsonStringify(allDeps, { space: 2 })
+      console.warn(`deps-dump - writing to ${writeAutoConfigDebug}`)
+      fs.writeFileSync(writeAutoConfigDebug, serialized)
```

Comment:
this part
```js
    // if writeAutoConfigDebug option enabled, dump allDeps to a file
    if (writeAutoConfigDebug) {
      const serialized = jsonStringify(allDeps, { space: 2 })
      console.warn(`deps-dump - writing to ${writeAutoConfigDebug}`)
      fs.writeFileSync(writeAutoConfigDebug, serialized)
    }
```

should appear in index.js or its own file. there should be a through or a spy added at the `'debug'`  pipeline phase

Reply:
doooone